api/src/main/java/org/hawkular/apm/api/model/trace/Node.java

```diff
@@ -324,16 +348,18 @@ public void addInteractionId(String id) {
      * @param scope The scope
      * @return The subset of correlation ids that are associated with the scope
      */
-    public List<CorrelationIdentifier> getCorrelationIds(Scope scope) {
+    public List<CorrelationIdentifier> findCorrelationIds(Scope... scope) {
         List<CorrelationIdentifier> ret = null;
 
         for (int i=0; i < correlationIds.size(); i++) {
             CorrelationIdentifier cid = correlationIds.get(i);
-            if (cid.getScope() == scope) {
-                if (ret == null) {
-                    ret = new ArrayList<CorrelationIdentifier>();
+            for (int j=0; j < scope.length; j++) {
+                if (cid.getScope() == scope[j]) {
+                    if (ret == null) {
+                        ret = new ArrayList<CorrelationIdentifier>();
```

Comment:
assing `new ArrayList<CorrelationIdentifier>()` on line 352 and remove ternary operator in return statement. Might be good to wrap the result into unmodifiable list